### PR TITLE
[BACKLOG-23730] As an admin I would like to have a user that could upload/download files using PUC outside of its home folder

### PR DESCRIPTION
**Pull request by neeta-rathi**
1. Created a new config property `download-roles` on `pentaho.xml`. This property holds a comma-separated list of roles that can download ( outside of their home folder ) content from repository
2. Added two end endpoints on `FileResource` -  `/canUpload` and `/canDownload`
3. These endpoints now calls `SystemUtils` utility methods `canUpload` and `canDownload` and defer the decision of uploading/downloading content

Per this new logic, rules for 

1. Downloading content:
If a user has permissions to "Read content" and "Create content", plus one of 2: "Administrator" OR "has a role listed on `<download-roles>`

2. Uploading content: ( no changes here )
If a user has permissions to "Read content" and "Create content", plus one of 2: "Administrator" OR "Publish Content"

@pentaho/rogueone

### CODE REVIEW

**Comment by pedrofvteixeira** [comment]
PR looks good. Please add UTs and/or ITs.

**Comment by neeta-rathi** [comment]
FileServiceTest accounts for new code placed on SystemUtils. I'm adding more tests for canUpload and canDownload endpoints as part of the next PR for calling these endpoint from the UI.

File: assemblies/pentaho-solutions/src/main/resources-filtered/pentaho-solutions/system/pentaho.xml [hunk 1]
```diff
@@ -23,6 +23,13 @@
       If true, show hint about sample users. Ultimately, should replace login-show-users-list.
     -->
     <login-show-sample-users-hint>true</login-show-sample-users-hint>
+
+	  <!--
+	  The download-roles defines user/roles that have download action permissions. It accepts a comma-separated list of roles.
+	  System admins must change what are the users/roles that have permissions for download action. If not defined here,
+	  the default value is set to 'Administrator'.
+	  -->
+	  <download-roles>Administrator</download-roles>
```
Comment (befc) [inline assemblies/pentaho-solutions/src/main/resources-filtered/pentaho-solutions/system/pentaho.xml]: Not sure, but indentation seems a bit different than the block above and bellow

**Comment by befc** [comment]
Apart from the Wingman results, everything seems fine

**Comment by neeta-rathi** [comment]
Closing this PR. New PR will be issued with UI work and additional unit tests!

**Pull request closed without merging**